src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentLoweringPass.cs

```diff
@@ -34,15 +34,21 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
             {
                 var reference = references[i];
                 var node = (TagHelperIntermediateNode)reference.Node;
+                if (node.TagHelpers.Any(t => t.IsChildContentTagHelper()))
+                {
+                    // Ignore, this will be handled when we rewrite the parent.
```

Comment:
I'd make the existing comment a little more clear to state that child content TagHelpers are dealt with elsewhere.